src/sysinfo/nvapi_adapter.h

```diff
@@ -25,10 +35,10 @@ namespace dxvk {
         [[nodiscard]] uint32_t GetPciBusId() const;
         [[nodiscard]] uint32_t GetPciDeviceId() const;
         [[nodiscard]] uint32_t GetBoardId() const;
-        [[nodiscard]] uint32_t GetVRamSize() const;
-        [[nodiscard]] uint32_t GetVirtualVRamSize() const;
         [[nodiscard]] std::optional<LUID> GetLuid() const;
         [[nodiscard]] NV_GPU_ARCHITECTURE_ID GetArchitectureId() const;
+        void UpdateVideoMemory();
```

Comment:
I somewhat don't like this, having to remember to call it before checking the values is extra mental load _and_ it makes it unable to query this if we ever got a const reference to an adapter. Can we instead persist only values that should be immutable for given adapter and query budget and current usage via some `[[nodiscard]] struct VideoMemoryInfo GetCurrentVideoMemoryInfo() const;`?

Reply:
yeah makes sense. Changed, though not yet completely happy with the naming.